logbook-jaxws/src/main/java/org/zalando/logbook/jaxws/LoggingHandler.java

```diff
@@ -0,0 +1,65 @@
+package org.zalando.logbook.jaxws;
+
+import java.io.IOException;
+import java.util.Set;
+
+import javax.xml.namespace.QName;
+import javax.xml.ws.handler.MessageContext;
+import javax.xml.ws.handler.soap.SOAPHandler;
+import javax.xml.ws.handler.soap.SOAPMessageContext;
+
+import org.zalando.logbook.Logbook;
+import org.zalando.logbook.Logbook.ResponseProcessingStage;
+
+public class LoggingHandler implements SOAPHandler<SOAPMessageContext> {
+
+    private static final ThreadLocal<ResponseProcessingStage> STAGE_CONTEXT = new ThreadLocal<>();
+    private Logbook logbook;
+
+    public LoggingHandler(Logbook logbook) {
+        this.logbook = logbook;
+    }
+
+    @Override
+    public boolean handleMessage(SOAPMessageContext context) {
+        boolean outbound = (boolean) context.get(MessageContext.MESSAGE_OUTBOUND_PROPERTY);
+
+        if (outbound) {
+            LocalRequest request = new LocalRequest(context);
+            try {
+                ResponseProcessingStage process = logbook.process(request).write();
+                STAGE_CONTEXT.set(process);
```

Comment:
Can't you put the stage into the context? That would remove the need to have a ThreadLocal here.